### ISSUE TITLE
Add support for pg_catalog.pg_type

### ISSUE THREAD

**Issue by awang** [MaterializeInc/materialize#4518]
npgsql (.net client library) complains about missing the `pg_catalog.pg_type` table. 

I'm putting this in the 0.6 milestone so we can track it, but after thinking about this more, ~I think this is low priority, since it's non-fatal and can be safely ignored~.  So I think we should do this after #3727 is complete, and only if it's reasonably straightforward.  (Assuming my assumptions remain true).  

cc @JLDLaughlin

**Comment by JLDLaughlin** [comment]
The good news is that this is also required for PgJDBC support, so I'll be working on this soon! (https://github.com/MaterializeInc/materialize/issues/4474, https://github.com/MaterializeInc/materialize/issues/4473, https://github.com/MaterializeInc/materialize/issues/4471, https://github.com/MaterializeInc/materialize/issues/3590)

**Comment by awang** [comment]
Update -- we think this is a blocker, so we're upgrading the priority of this

**Comment by JLDLaughlin** [comment]
Talking with Nikhil, this issue is actually blocked by https://github.com/MaterializeInc/materialize/issues/4543

**Comment by benesch** [comment]
Npgsql and PgJDBC both are very stringent in what they require from `pg_type`. We're going to temporarily hit pause on this issue to evaluate our options here—mostly by looking at what CockroachDB does. @JLDLaughlin is going to start drafting a design for #4460 in the meantime.
